File: src/main/java/ar/edu/unq/spring/persistence/repository/jpa/ItemDAO.java
package ar.edu.unq.spring.persistence.repository.jpa;

import ar.edu.unq.spring.modelo.Item;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.Set;

@Repository
public interface ItemDAO extends JpaRepository<Item, Long> {

    Item findTopByOrderByPesoDesc();

    @Query(
            "FROM Item i where i.peso  > :peso order by i.peso asc"
    )
    Set<Item> getMasPesados(@Param("peso") int peso);

    @Query(
            "from Item i "
                    + "order by i.peso asc"
    )
    Set<Item> getItemsDePersonajesDebiles(int vida);
}

File: src/main/kotlin/ar/edu/unq/spring/ItemDAO.kt
package ar.edu.unq.spring

import ar.edu.unq.spring.modelo.Item
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query

interface ItemDAO : JpaRepository<Item, Long> {

    fun findTopByOrderByPesoDesc(): Item

    @Query(
        "from Item i "
                + "where i.owner.vida < ?1 "
                + "order by i.peso asc"
    )
    fun getItemsDePersonajesDebiles(unaVida: Int): Collection<Item>


    @Query(
        "FROM Item i where i.peso  > ?1 order by i.peso asc"
    )
    fun getMasPesados(unValorDado: Int): Collection<Item>

}
